src/Microsoft.Azure.SignalR/EndpointProvider/ServiceScaleManager.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal class ServiceScaleManager
+    {
+        private readonly ILogger _logger;
+        private IReadOnlyList<ServiceEndpoint> _endpointsStore;
+
+        public ServiceScaleManager(ILoggerFactory loggerFactory,
+            IOptionsMonitor<ServiceOptions> optionsMonitor)
+        {
+            _logger = loggerFactory?.CreateLogger<ServiceScaleManager>();
+
+            // Disable optionsMonitor until feature ready.
+            // optionsMonitor.OnChange(OnChange);
+
+            _endpointsStore = optionsMonitor.CurrentValue.Endpoints;
+        }
+
+        private void OnChange(ServiceOptions options)
```

Comment:
If the configuration is changed intensively during some period... shall the new detected Change cancel the previous waiting and begin the latest sync?

Reply:
For this async endpoints change, it's not good to cancel previous operation and hard to rollback some changes. This may lead to some available endpoints inconsistent in `ServiceEndpointManager` and `MultiEndpointServiceConnectionContainer` with some unexpected behaviors.
* For add, probably only final add negotiation is waiting and can be cancelled, but some server connections are already established.
* For remove, offline call is triggered to drop clients and remove server routing, probably only server connection stop and clean is able to cancel. 
So I think it's good to respect every time configuration change. App server should be aware of the impact of value update.